### PR TITLE
MULE-16205: IdempotentRedeliveryPolicy blocks CPU_LITE thread

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: integration/src/test/java/org/mule/test/components/RedeliveryPolicyTestCase.java [hunk 1]
```diff
@@ -32,16 +49,54 @@ protected String getConfigFile() {
...
   }
 
+  private void sendDataWeaveObjectMessage(String flowName) throws Exception {
+    flowRunner(flowName)
+        .withPayload("{ \"name\" : \"bruce\"}")
+        .withMediaType(MediaType.APPLICATION_JSON)
+        .run();
+  }
+
+  private static CountDownLatch latch;
```
Comment (marianogonzalez) [inline integration/src/test/java/org/mule/test/components/RedeliveryPolicyTestCase.java]: static fields and classes should be defined before instance members

File: integration/src/test/java/org/mule/test/components/RedeliveryPolicyTestCase.java [hunk 2]
```diff
@@ -32,16 +49,54 @@ protected String getConfigFile() {
...
+  private static AtomicInteger awaiting = new AtomicInteger();
```
Comment (marianogonzalez) [inline integration/src/test/java/org/mule/test/components/RedeliveryPolicyTestCase.java]: static fields should be upper case

File: integration/src/test/java/org/mule/test/components/RedeliveryPolicyTestCase.java [hunk 3]
```diff
@@ -6,23 +6,40 @@
...
-  public void setUp() {
+  public void before() {
     queueHandler = new TestConnectorQueueHandler(registry);
+    latch = new CountDownLatch(1);
+    awaiting.set(0);
+  }
+
+  @After
+  public void after() throws Exception {
+    latch.countDown();
```
Comment (marianogonzalez) [inline integration/src/test/java/org/mule/test/components/RedeliveryPolicyTestCase.java]: what's this for? looks like a left over

File: integration/src/test/java/org/mule/test/components/RedeliveryPolicyTestCase.java (hunk 2 above)
Comment (elrodro83) [reply integration/src/test/java/org/mule/test/components/RedeliveryPolicyTestCase.java]: only if they were `final`

File: integration/src/test/java/org/mule/test/components/RedeliveryPolicyTestCase.java (hunk 3 above)
Comment (elrodro83) [reply integration/src/test/java/org/mule/test/components/RedeliveryPolicyTestCase.java]: updated the test

**Pull request merged**